skorch/net.py

```diff
@@ -1251,19 +1251,48 @@ def get_iterator(self, dataset, training=False):
     def _get_params_for(self, prefix):
         return params_for(prefix, self.__dict__)
 
-    def _get_params_for_optimizer(self, prefix, named_parameters):
-        """Parse kwargs configuration for the optimizer identified by
-        the given prefix. Supports param group assignment using wildcards:
+    def get_params_for(self, prefix):
+        """Collect and return init parameters for an attribute.
 
-            optimizer__lr=0.05,
-            optimizer__param_groups=[
-                ('rnn*.period', {'lr': 0.3, 'momentum': 0}),
-                ('rnn0', {'lr': 0.1}),
-            ]
+        Attributes could be, for instance, pytorch modules, criteria,
+        or data loaders (for optimizers, use
+        :meth:`.get_params_for_optimizer` instead). Use the returned
```

Comment:
There is still a use case for using `get_params_for` with an optimizer as shown in the change in `examples/translation/translation.ipynb` in this PR.

Reply:
The change I made there was just a simple search and replace. Now, after actually looking at this, I believe it should use `_get_params_for_optmizer` instead. Or is there any reason not to?

Reply:
Yea it should use `get_params_for_optimizer` just in case someone uses glob matching for `param_groups`?

Reply:
I changed the code to use `get_params_for_optimizer`.